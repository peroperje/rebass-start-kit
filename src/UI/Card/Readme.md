Wrapper for [Rebass Card Component](https://rebassjs.org/card)

```js
<Card>
  <div>This is card content</div>
</Card>
```
