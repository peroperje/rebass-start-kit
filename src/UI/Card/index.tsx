import * as rebass from 'rebass';

const Card = rebass.Card;
export default Card;
