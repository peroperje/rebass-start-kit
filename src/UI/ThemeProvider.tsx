import React from "react";
import { ThemeProvider as EmotionThemeProvider } from "emotion-theming";

import theme from "./theme";

export interface Props {
  children: React.ReactNode;
}

const ThemeProvider: React.FC<Props> = ({ children }: Props): JSX.Element => (
  <EmotionThemeProvider theme={theme}>{children}</EmotionThemeProvider>
);

export default ThemeProvider;
