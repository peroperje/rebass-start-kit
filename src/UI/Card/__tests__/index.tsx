import React from 'react';
import render from 'react-test-renderer';

import ThemeProvider from '../../ThemeProvider';

import Card from '../index';

describe('Card', () => {
    it('Default snapshot', () => {
        const tree = render
            .create(
                <ThemeProvider>
                    <Card>Test</Card>
                </ThemeProvider>,
            )
            .toJSON();
        expect(tree).toMatchSnapshot();
    });
});
