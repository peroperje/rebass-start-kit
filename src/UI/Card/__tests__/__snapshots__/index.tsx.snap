// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card Default snapshot 1`] = `
.emotion-0 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  padding: 8px;
  background-color: #fff;
  box-shadow: 0 0 4px rgba(0,0,0,.125);
}

<div
  className="emotion-0"
>
  Test
</div>
`;
